### ISSUE TITLE
Error loading an img-file with GdalImageProvider

### ISSUE THREAD

**Issue by jany-tenaj** [DotSpatial/DotSpatial#807]
I tried to load an image with the GdalImageProvider. It was added to the legend but the image was not visible.

### Steps to reproduce

1. Run DemoMap. 
2. Add an *.img - file via the Add Layer... button.

DotSpatial version: master

![noimage](https://cloud.githubusercontent.com/assets/13728699/15929256/63f5c10a-2e4c-11e6-94ab-e4fc7ff8c43d.png)

### Expected behaviour

Image should be visible.

### Actual behaviour

Image is not visible.

I had a look at this problem and found an error in line [556](https://github.com/DotSpatial/DotSpatial/blob/2e0f860d5160d720da85c25e105ff21ed92cc382/Source/DotSpatial.Data.Rasters.GdalExtension/GdalImage.cs#L556) inside **ReadPaletteBuffered(int xOffset, int yOffset, int xSize, int ySize, Band first)** inside **DotSpatial.Data.Rasters.GdalExtension.GdalImage.cs**.

```
private Bitmap ReadPaletteBuffered(int xOffset, int yOffset, int xSize, int ySize, Band first)
        {
 ... omitted code ...
 for (int row = 0; row < height; row++)
            {
                for (int col = 0; col < width; col++)
                {
                    Array.Copy(colorTable[r[col + row * stride]], 0, vals,
                               row * stride + col * bpp, 4);
                }
            }
            Marshal.Copy(vals, 0, bData.Scan0, vals.Length);
            result.UnlockBits(bData);
            return result;
}
```
If we replace stride with width as it is in ReadPaletteBuffered() the function no longer causes an error. 
But the resulting image looks nothing like it should. When zoomed to full extent it looks like a small vertical stripe instead of the square it normally is. 
When zooming in its size changes to different rectangles that make no sense. Furthermore you can see small vertical white stripes between the colored parts that were loaded.

Unfortunately I'm not allowed to attach the img file.

**Comment by mogikanin** [comment]
@jany-tenaj I think there is a typo and correct code should be
```Array.Copy(colorTable[r[col + row * width]], 0, vals, row * stride + col * bpp, 4);```
The similar code also exists in [GdalTiledImage](https://github.com/DotSpatial/DotSpatial/blob/master/Source/DotSpatial.Data.Rasters.GdalExtension/GdalTiledImage.cs#L155).

**Comment by jany-tenaj** [comment]
@mogikanin Yes that's what I think, too. But as I said the error is gone if we use width but the resulting image looks bad. That's why I created this issue instead of simply replacing stride with width. 

Yesterday I had an image that had 3 color bands and I was able to load this with GdalImageProvider.ReadRGB() and have it look correct. So I think there is something fishy inside ReadPaletteBuffered(int xOffset, int yOffset, int xSize, int ySize, Band first).

**Comment by mogikanin** [comment]
@jany-tenaj , I've fixed typo in Array.Copy(...). Regarding to incorrect showing. Maybe you can change a bit img file, and attach it, otherwise this issue has no sense without source data :)

**Comment by jany-tenaj** [comment]
@mogikanin yes I know that it would be good to attach the file. But as I said I'm not allowed. As I need to fix this issue for my own program anyway I'll have a look at it as soon as I have enough time to continue fixing DS errors. 

Looking at the description of #824 I'd say this could be related.
